test/e2e/shoot/create_rotate-ca_delete.go

```diff
@@ -146,3 +165,83 @@ func verifyCABundleInKubeconfigSecret(ctx context.Context, g Gomega, c client.Re
 	g.Expect(c.Get(ctx, shootKey, secret)).To(Succeed())
 	g.Expect(secret.Data).To(HaveKeyWithValue("ca.crt", expectedBundle))
 }
+
+type secretConfigNamesToSecrets map[string][]corev1.Secret
+
+func groupByName(allSecrets []corev1.Secret) secretConfigNamesToSecrets {
+	grouped := make(secretConfigNamesToSecrets)
+	for _, secret := range allSecrets {
+		grouped[secret.Labels["name"]] = append(grouped[secret.Labels["name"]], secret)
+	}
+
+	// sort by age
+	for _, secrets := range grouped {
+		sort.Slice(secrets, func(i, j int) bool {
+			return secrets[i].CreationTimestamp.Before(&secrets[j].CreationTimestamp)
+		})
+	}
+	return grouped
+}
+
+var managedByProviderLocalSecretsManager = client.MatchingLabels{
+	"managed-by":       "secrets-manager",
+	"manager-identity": "provider-local-controlplane",
+}
+
+const (
+	caProviderLocalControlPlane       = "ca-provider-local-controlplane"
+	caProviderLocalControlPlaneBundle = "ca-provider-local-controlplane-bundle"
+	providerLocalDummyServer          = "provider-local-dummy-server"
+	providerLocalDummyAuth            = "provider-local-dummy-auth"
+)
+
+func verifyProviderLocalSecretsBefore(ctx context.Context, g Gomega, c client.Reader, namespace string) secretConfigNamesToSecrets {
+	secretList := &corev1.SecretList{}
+	g.Expect(c.List(ctx, secretList, client.InNamespace(namespace), managedByProviderLocalSecretsManager)).To(Succeed())
+
+	grouped := groupByName(secretList.Items)
+	g.Expect(grouped).To(And(
+		HaveKeyWithValue(caProviderLocalControlPlane, HaveLen(1)),
+		HaveKeyWithValue(caProviderLocalControlPlaneBundle, HaveLen(1)),
+		HaveKeyWithValue(providerLocalDummyServer, HaveLen(1)),
+		HaveKeyWithValue(providerLocalDummyAuth, HaveLen(1)),
+	), "all secrets should get created, but not rotated yet")
+
+	return grouped
+}
+
+func verifyProviderLocalSecretsPrepared(ctx context.Context, g Gomega, c client.Reader, namespace string, secretsBefore secretConfigNamesToSecrets) secretConfigNamesToSecrets {
+	secretList := &corev1.SecretList{}
+	g.Expect(c.List(ctx, secretList, client.InNamespace(namespace), managedByProviderLocalSecretsManager)).To(Succeed())
+
+	grouped := groupByName(secretList.Items)
+	g.Expect(grouped).To(And(
+		HaveKeyWithValue(caProviderLocalControlPlane, HaveLen(2)),
+		HaveKeyWithValue(caProviderLocalControlPlaneBundle, HaveLen(1)),
+		HaveKeyWithValue(providerLocalDummyServer, HaveLen(1)),
```

Comment:
Do you also want to add a client cert here and check that it already got renewed?